### ISSUE TITLE
The visiting order of nodes in CFG

### ISSUE THREAD

**Issue by jianghaizhi** [angr/angr#1185]
Hello, all,

I can get a list of the nodes  in the CFG by this code: `cfg.graph.nodes`
But I do not know what's the visiting order of the nodes in this list.
Is it a depth-first or breadth-first search in CFG？

If I want to get a list of all the nodes in CFG and follow the order of depth-first search, May I know how to write the code?

Thank you so much.

**Comment by rhelmot** [comment]
There is not an order to this list, it is probably sorted by python's hash function or something.

`cfg.graph` is a DiGraph object from the NetworkX library. If you would like to produce a list or nodes in some traversal order, you should look up some examples using NetworkX to do that.

**Comment by jianghaizhi** [comment]
Thank you @rhelmot .

I checked the document of angr, It seems that the implantation of `nodes()` is in a traversal order, but what kind of the order is not specified in the document.

I am new to Angr and not sure if I understand this correctly.

<img width="740" alt="screen shot 2018-08-26 at 19 58 55" src="https://user-images.githubusercontent.com/9067035/44634708-3cb6ae00-a96b-11e8-800b-66bb1e26b2e8.png">

**Comment by rhelmot** [comment]
You're looking at the wrong part of the docs. That applies to a different class from `cfg.graph`. You should use `print(type(cfg.graph))` to learn what class you're working with, and then look up documentation for that class. (it is not an angr class, but a networkx class)

**Comment by jianghaizhi** [comment]
Thank you so much. @rhelmot 
Is it available to use the functions nodes() and sort_nodes() in class angr.analyses.forward_analysis.GraphVisitor to get a depth-first list of all the nodes in CFG? May I know how to write the code?

**Comment by rhelmot** [comment]
No, you can't. GraphVisitor is meant for designing data-flow analyses.

Use networkx, which is designed for this task. cfg.graph is a networkx digraph.
